iphone/Classes/TiUIWebView.m

```diff
@@ -707,24 +719,25 @@ - (BOOL)webView:(UIWebView *)webView shouldStartLoadWithRequest:(NSURLRequest *)
 	BOOL allHeadersIncluded = NO;
 	int requiredHeaders = (int)[newHeaders count];
     
-	for (NSString* existingHeader in [[request allHTTPHeaderFields] allKeys]) {
-		for (NSString* newHeader in [newHeaders allKeys]) {
-			if ([[existingHeader lowercaseString] isEqualToString:[newHeader lowercaseString]]) {
-				requiredHeaders--;
+	if (newHeaders) {
+		for (NSString* existingHeader in [[request allHTTPHeaderFields] allKeys]) {
+			for (NSString* newHeader in [newHeaders allKeys]) {
+				if ([[existingHeader lowercaseString] isEqualToString:[newHeader lowercaseString]]) {
+					requiredHeaders--;
+				}
 			}
 		}
-	}
-    
-	// If there are custom headers and not all of them are included, add them
-	if (newHeaders != nil && requiredHeaders > 0) {
-		NSMutableURLRequest* newRequest = [request mutableCopy];
-		for (NSString* newHeader in [newHeaders allKeys]) {
-			[newRequest addValue:[newHeaders valueForKey:newHeader] forHTTPHeaderField:newHeader];
+
+		if (requiredHeaders > 0) {
+			NSMutableURLRequest* newRequest = [request mutableCopy];
+			for (NSString* newHeader in [newHeaders allKeys]) {
+				[newRequest addValue:[newHeaders valueForKey:newHeader] forHTTPHeaderField:newHeader];
+			}
+			[self loadURLRequest:newRequest];
+			[newRequest release];
+
+			return NO;
 		}
-		[self loadURLRequest:newRequest];
-		[newRequest release];
-        
-		return NO;
 	}
```

Comment:
Note 2/2: I moved the `requestHeaders` logic into the `newHeaders` scope, so it doesn't affect requests without custom request headers. This fixes a possible glitch in TIMOB-24329